gcp_org_resource_count.sh

```diff
@@ -0,0 +1,158 @@
+
+while [[ $# -gt 0 ]]
+do
+    key="$1"
+    case $key in
+        -g|--organization_id)
+        ORG_ID="$2"
+        shift # past argument
+        shift # past value
+        ;;
+        *)    # unknown option
+        shift # past argument
+        ;;
+    esac
+done
+
+if [ -z "$ORG_ID" ]; then
+    echo "[+] Getting organizations list"
+    ORGS=$(gcloud organizations list --format="value(name)")
+    ORGS_LEN=$(echo "$ORGS" | wc -l)
+    echo "Found ${ORGS_LEN} organizations"
+    echo "[+] Iterating organizations and counting resources"
+else
+    ORGS="$ORG_ID"
+    echo "[+] counting resources for organizations id ${ORG_ID}"
+fi
+echo
+
+counter=0
+total_vms=0
+total_functions=0
+total_cloud_run=0
+total_container_images=0
+total_vm_images=0
+total_gke_nodes=0
+total_buckets=0
+total_private_buckets=0
+active_user=$(gcloud config get-value account)
+
+# listed here https://cloud.google.com/asset-inventory/docs/asset-types
+asset_types="compute.googleapis.com/Instance,cloudfunctions.googleapis.com/CloudFunction,run.googleapis.com/Service,containerregistry.googleapis.com/Image,compute.googleapis.com/Image,k8s.io/Node,storage.googleapis.com/Bucket"
+
+for org_id in $ORGS; do
+  echo "Processing organization: $org_id"
+
+  all_resources=$(gcloud asset search-all-resources --scope="organizations/$org_id" --format="json(displayName, assetType, project)" --asset-types="$asset_types")
+  curr_vms=$(echo "$all_resources" | jq -r '.[] | select(.assetType == "compute.googleapis.com/Instance") | .displayName' | wc -l)
+  curr_functions=$(echo "$all_resources" | jq -r '.[] | select(.assetType == "cloudfunctions.googleapis.com/CloudFunction") | .displayName' | wc -l)
+  curr_cloud_run=$(echo "$all_resources" | jq -r '.[] | select(.assetType == "run.googleapis.com/Service") | .displayName' | wc -l)
+  curr_container_images=$(echo "$all_resources" | jq -r '.[] | select(.assetType == "containerregistry.googleapis.com/Image") | .displayName' | wc -l)
+  curr_vm_images=$(echo "$all_resources" | jq -r '.[] | select(.assetType == "compute.googleapis.com/Image") | .displayName' | wc -l)
+  curr_gke_nodes=$(echo "$all_resources" | jq -r '.[] | select(.assetType == "k8s.io/Node") | .displayName' | wc -l)
+  curr_buckets=$(echo "$all_resources" | jq -r '.[] | select(.assetType == "storage.googleapis.com/Bucket") | .displayName' | wc -l)
+
+  total_vms=$((total_vms + curr_vms))
+  total_functions=$((total_functions + curr_functions))
+  total_cloud_run=$((total_cloud_run + curr_cloud_run))
+  total_container_images=$((total_container_images + curr_container_images))
+  total_vm_images=$((total_vm_images + curr_vm_images))
+  total_gke_nodes=$((total_gke_nodes + curr_gke_nodes))
+  total_buckets=$((total_buckets + curr_buckets))
+
+
+
+  is_public_access_prevention_enabled_from_org_level=$(gcloud resource-manager org-policies describe storage.publicAccessPrevention --organization="$org_id" --effective --format json |  jq '.booleanPolicy.enforced == true')
```

Comment:
can you please align the logic with what we currently have in the check_public function in GCP client? I think we check only for project-level permissions and we don't want it to be very different from what we have in the scan logic